flood_adapt/dbs_classes/dbs_benefit.py

```diff
@@ -0,0 +1,94 @@
+from flood_adapt.object_model.interface.benefits import IBenefit
+
+
+class DbsBenefit(DbsTemplate):
+    _type = "benefit"
+    _folder_name = "benefits"
+    _object_model_class = Benefit
+
+    def save(self, benefit: IBenefit, overwrite: bool = False):
+        """Saves a benefit object in the database.
+
+        Parameters
+        ----------
+        measure : IBenefit
+            object of scenario type
+        overwrite : bool, optional
+            whether to overwrite existing benefit with same name, by default False
+
+        Raises
+        ------
+        ValueError
+            Raise error if name is already in use. Names of benefits assessments should be unique.
+        """
+
+        # Check if all scenarios are created
+        if not all(benefit.scenarios["scenario created"] != "No"):
+            raise ValueError(
+                f"'{benefit.attrs.name}' name cannot be created before all necessary scenarios are created."
+            )
+
+        # Save the benefit
+        super().save(benefit, overwrite=overwrite)
+
+    def delete(self, name: str, toml_only: bool = False):
+        """Deletes an already existing benefit in the database.
+
+        Parameters
+        ----------
+        name : str
+            name of the benefit
+        toml_only : bool, optional
+            whether to only delete the toml file or the entire folder. If the folder is empty after deleting the toml,
+            it will always be deleted. By default False
+
+        Raises
+        ------
+        ValueError
+            Raise error if benefit has already model output
+        """
+
+        # First delete the benefit
+        super().delete(name, toml_only=toml_only)
+
+        # Delete output if edited
+        output_path = self._database.input_path.parent / "output" / "Benefits" / name
+
+        if output_path.exists():
+            shutil.rmtree(output_path, ignore_errors=True)
+
+    def edit(self, benefit: IBenefit):
+        """Edits an already existing benefit in the database.
+
+        Parameters
+        ----------
+        benefit : IBenefit
+            benefit to be edited in the database
+
+        Raises
+        ------
+        ValueError
+            Raise error if name is already in use.
+        """
+        # Check if it is possible to edit the benefit. This then also covers checking whether the
+        # benefit is already used in a higher level object. If this is the case, it cannot be edited.
+        try:
+            super().edit(benefit)
+        except ValueError as e:
+            # If not, raise error
+            raise ValueError(e)
+        else:
+            # Delete output if edited
+            output_path = (
+                self._database.input_path.parent
+                / "output"
+                / "Benefits"
+                / benefit.attrs.name
+            )
+
+            if output_path.exists():
+                shutil.rmtree(output_path, ignore_errors=True)
```

Comment:
I didnt know you could do try except else, seems strange but it works so nice one!

Reply:
Now that I look at it again, I also don't really know what my idea was here. The only thing it adds now is that it still tries to save it when you get an error other than a ValueError. Which doesn't even feel right haha

Reply:
And the locked behaviour can be removed. Requirements now are that if something is in a scenario, it cannot be deleted anyway. So this functionality is obsolete